network-proxy-service/src/main.rs

```diff
@@ -0,0 +1,33 @@
+pub mod interface;
+pub mod sshforwarding;
+pub mod errors;
+pub mod logging;
+pub mod networkproxy;
+
+use std::io::{self, Write};
+
+use interface::NetworkProxyConfig;
+use logging::{init_tracing, Must};
+
+#[tokio::main]
+async fn main() -> io::Result<()> {
+    init_tracing();
+    let proxy_config: NetworkProxyConfig = serde_json::from_reader(io::stdin()).or_bail("valid input.");
+    let mut proxy = proxy_config.new_proxy();
+
+    proxy.prepare().await.or_bail("Failed to prepare network proxy.");
```

Comment:
Good point. Yes, it is. 

Here is the sequence of the steps:
1. `local_listener` is binded to a port during `proxy.prepare()`. And it able to accept connections.
2. `READY` message is sent back to go side.
3. `local_listener` start to process received requests after `proxy.start_serve` is called, and will keep serving until stopped by external signals.

Any request coming during step 2 and step 3 is not rejected. It simply waits for a tiny amount of time until the `local_listener.accept` is called and start being processed.

Since the number of requests to the proxy is small, and the gap between steps 2 and 3 is small, I feel it is safe to go with this sequence.

Reply:
Done!